OCPQE-15832: Migrate 84_IPI on AZURE file-csi to prow

Phaow opened the pull request:
## Migrate azurefile csi ci profiles from jenkins to Prow
- [OCPQE-15832](https://issues.redhat.com/browse/OCPQE-15832) Migrate 84_IPI on AZURE & Empty Resource Group & OVN IPSEC & Azure-file-csi to Prow

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-azure-ipi-ovn-ipsec-file-p1-f14

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-azure-ipi-ovn-ipsec-file-p2-f7

Phaow commented:
Hi @jianlinliu @liangxia , could you please help take a look for the workflow and jobs changed part when you get a chance? The new workflow just change the default storageclass to `azurefile-csi`, it **[worked](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/40959/rehearse-40959-periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-azure-ipi-ovn-ipsec-file-p1-f14/1676502585462231040/artifacts/azure-ipi-ovn-ipsec-file-p1-f14/storage-conf-storageclass-set-default-storageclass/build-log.txt)** as expected. Thanks in advance!
cc to @duanwei33

jianlinliu commented on ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/ovn/ipsec/file/provision/cucushift-installer-rehearse-azure-ipi-ovn-ipsec-file-provision-chain.metadata.json:
```diff
@@ -0,0 +1,19 @@
+{
+	"path": "cucushift/installer/rehearse/azure/ipi/ovn/ipsec/file/provision/cucushift-installer-rehearse-azure-ipi-ovn-ipsec-file-provision-chain.yaml",
```
The keyword `file` is too common, not easy to understand, how about `filestore-csi`, or something else?

Phaow replied:
The full name of the csi driver is `azurefile-csi-driver`, but seems azure is a bit duplicated, I'll rename as `file-csi` instead. Thanks!

duanwei33 replied:
I think we can call it `azurefile` or `azurefile-csi`, `file-csi` sounds a bit strange.

Phaow replied:
Sure, but the chain or workflow name already contains the `azure`, `cucushift-installer-rehearse-azure-ipi-ovn-ipsec-azurefile` , is it a bit duplicate, also the job name has some limitation about the length, we'd better make the name as shorter as we can also make it more clearly?

Phaow replied:
Sure, thanks for the double check, let me update it to `azurefile-csi`.

jianlinliu commented on ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/ovn/ipsec/file/provision/cucushift-installer-rehearse-azure-ipi-ovn-ipsec-file-provision-chain.yaml:
```diff
@@ -0,0 +1,14 @@
+  - ref: ovn-conf
+  - ref: ovn-conf-ipsec-manifest
+  - chain: ipi-install
+  - ref: enable-qe-catalogsource
```
seem like the same steps as `cucushift-installer-rehearse-azure-ipi-ovn-ipsec-provision`, so can we directly create the workflow yaml file to call `cucushift-installer-rehearse-azure-ipi-ovn-ipsec-provision` instead of adding this new provision chain file? just like what we did in `./gcp/ipi/filestore-csi/cucushift-installer-rehearse-gcp-ipi-filestore-csi-workflow.yaml`

Phaow replied:
Sure, thanks I forgot update it.

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-azure-ipi-ovn-ipsec-file-p1-f14

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-azure-ipi-ovn-ipsec-file-p1-f14

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-ovn-ipsec-azurefile-csi-p1-f14

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-azure-ipi-ovn-ipsec-azurefile-csi-p2-f7

Phaow commented:
Hi @jianlinliu @liangxia , addressed the comments, could you please help take a look for the workflow and jobs changed part again when you get a chance? The new workflow just change the default storageclass to azurefile-csi, it [worked](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/40959/rehearse-40959-periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-azure-ipi-ovn-ipsec-azurefile-csi-p2-f7/1676822726293065728/artifacts/azure-ipi-ovn-ipsec-azurefile-csi-p2-f7/storage-conf-storageclass-set-default-storageclass/build-log.txt) as expected. Thanks in advance!

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-ovn-ipsec-azurefile-csi-p1-f14

jianlinliu commented on ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/ovn/ipsec/azurefile-csi/provision/cucushift-installer-rehearse-azure-ipi-ovn-ipsec-azurefile-csi-provision-chain.yaml:
```diff
@@ -0,0 +1,10 @@
+chain:
+  as: cucushift-installer-rehearse-azure-ipi-ovn-ipsec-azurefile-csi-provision
```
I suggest to directly copy the steps of the chain to workflow yaml file

Phaow replied:
Addressed, thanks for the suggestions!

jianlinliu commented:
/lgtm

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-ovn-ipsec-azurefile-csi-p1-f14

liangxia commented:
/approve

Phaow commented:
/pj-rehearse ack
